rs/src/lib.rs

```diff
@@ -59,6 +59,10 @@ pub struct InvalidMethod;
 #[derive(Debug, Clone)]
 pub struct InvalidScheme;
 
+/// Indicates an IP address could not be decoded.
+#[derive(Debug, Clone)]
+pub struct InvalidIpAddress;
```

Comment:
Nit, take it or leave it: AFAICT this is only returned for IPv6 addrs. Do you think the error (or its formatted message) should indicate that?

Reply:
It's also used when an IP address is not provided in a tcp address

Reply:
missed that, nvm